### DISCUSSION TITLE
Trouble building server, CMake stuck downloading mariadb connector/c++

### DISCUSSION THREAD

**Discussion by 0utta-Ctrl** [DarkflameUniverse/DarkflameServer#1297] [Q&A] [answered]
All,

When building the server, everything goes fine until cmake tries to download the mariadb c++ connector.  I have been patient but it shouldn't take hours to download this.  I am running cmake 3.18, Visual Studio 2022, and MariaDB 11.1 on Windows 10.

Any help would be appreciated.

Here's the tail end of my terminal:

1> Command line: "C:\Windows\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_BUILD_TYPE:STRING="RelWithDebInfo"   "E:\Lego Universe\DFU\DarkflameServer" 2>&1"
1> Working directory: E:/Lego Universe/DFU/DarkflameServer/build
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.37.32825.0
1> [CMake] -- The CXX compiler identification is MSVC 19.37.32825.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Variable: PROJECT_VERSION_MAJOR = 1
1> [CMake] -- Variable: PROJECT_VERSION_MINOR = 1
1> [CMake] -- Variable: PROJECT_VERSION_PATCH = 1
1> [CMake] -- Variable: LICENSE = AGPL-3.0
1> [CMake] -- Variable: __dynamic = 1
1> [CMake] -- Variable: __maria_db_connector_compile_jobs__ = 1
1> [CMake] -- Variable: __enable_testing__ = 1
1> [CMake] -- Variable: OPENSSL_ROOT_DIR = /usr/local/opt/openssl@3/
1> [CMake] -- Version: 1.1.1
1> [CMake] -- Checking resource file integrity
1> [CMake] -- Moved sharedconfig.ini to project binary directory
1> [CMake] -- Moved authconfig.ini to project binary directory
1> [CMake] -- Moved chatconfig.ini to project binary directory
1> [CMake] -- Moved worldconfig.ini to project binary directory
1> [CMake] -- Moved masterconfig.ini to project binary directory
1> [CMake] -- Moved blacklist.dcf to project binary directory
1> [CMake] -- Resource file integrity check complete
1> [CMake] Downloading mariadb connector/c
1> [CMake] Downloading mariadb connector/c++

**Comment by EmosewaMC** [comment]
`https://dlm.mariadb.com/2531525/Connectors/cpp/connector-cpp-1.0.2/mariadb-connector-cpp-1.0.2-win64.msi`
This is the expanded download link, are you able to download this?  If so, can you check if you can do so on at least 2 browsers?

**Comment by EmosewaMC** [comment]
![image](https://github.com/DarkflameUniverse/DarkflameServer/assets/39972741/241cc572-2d85-4406-8071-27efbb34d0ad)
Here is what the file name and size is

**Comment by 0utta-Ctrl** [comment]
Thank you, I resolved this issue by removing spaces in the path to the DarkflameServer file.

**Comment by EmosewaMC** [comment, marked as answer]
We now have officially supported releases that require little setup and do _not_ require building from source.  Please use those located [here](https://github.com/DarkflameUniverse/DarkflameServer/releases)
